packages/react-components/src/components/TroubleShootingGuideErrorBar.tsx

```diff
@@ -0,0 +1,150 @@
+   * experienced when connecting to a call.
+   *
+   * @example
+   * ```ts
+   * onNetworkingTroubleshootingClick?: () =>
+   *  window.open('https://contoso.com/network-troubleshooting', '_blank');
+   * ```
+   *
+   * @remarks
+   * if this is not supplied, the composite will not show a 'network troubleshooting' link.
+   */
+  onNetworkingTroubleshootingClick?: () => void;
+  /**
+   * strings related to trouble shooting guidance link and dismiss button
+   */
+  troubleshootingGuideStrings: _TroubleshootingGuideErrorBarStrings;
+}
+
+/**
+ * @internal
+ * A component to show device Permission/network connection related errors, contains link that leads to further trouble shooting guide
+ */
+export const _TroubleshootingGuideErrorBar = (props: _TroubleshootingGuideErrorBarProps): JSX.Element => {
+  const theme = useTheme();
+  // error bar strings
+  const localeStrings = useLocale().strings.errorBar;
+  const strings = props.strings ?? localeStrings;
+
+  const [dismissedErrors, setDismissedErrors] = useState<DismissedError[]>([]);
+
+  const {
+    onPermissionsTroubleshootingClick,
+    onNetworkingTroubleshootingClick,
+    permissionsState,
+    troubleshootingGuideStrings
+  } = props;
+
+  // dropDismissalsForInactiveErrors only returns a new object if `dismissedErrors` actually changes.
+  // Without this behaviour, this `useEffect` block would cause a render loop.
+  useEffect(
+    () => setDismissedErrors(dropDismissalsForInactiveErrors(props.activeErrorMessages, dismissedErrors)),
+    [props.activeErrorMessages, dismissedErrors]
+  );
+
+  const toShow = errorsToShow(props.activeErrorMessages, dismissedErrors);
+
+  return (
+    <Stack data-ui-id="error-bar-stack">
+      {toShow.map((error) => (
+        <MessageBar
+          {...props}
+          styles={messageBarStyle(theme, messageBarType(error.type))}
+          key={error.type}
+          messageBarType={messageBarType(error.type)}
+          messageBarIconProps={messageBarIconProps(error.type)}
+        >
+          <Stack horizontal horizontalAlign="space-between" verticalAlign="center">
+            {onPermissionsTroubleshootingClick || onNetworkingTroubleshootingClick ? (
+              <div>
+                {strings[error.type]}
+                <Link
+                  styles={linkStyle(theme)}
+                  onClick={() => {
+                    if (onPermissionsTroubleshootingClick) {
+                      onPermissionsTroubleshootingClick(permissionsState);
+                    } else if (onNetworkingTroubleshootingClick) {
+                      onNetworkingTroubleshootingClick();
+                    }
+                  }}
+                  underline
+                >
+                  <span>{troubleshootingGuideStrings.linkText}</span>
+                </Link>
+              </div>
+            ) : (
+              <div>{strings[error.type]} </div>
+            )}
+            <PrimaryButton
```

Comment:
Can we move this into the "actions" prop?
https://developer.microsoft.com/en-us/fluentui#/controls/web/messagebar

Reply:
fixed that!